02/app.js

```diff
@@ -2,82 +2,82 @@
 
 var username = prompt('Welcome! What is your name?');
 alert('Hi, ' + username + ', are you ready to take the BrodyQuiz?');
-
 var strQ = ['Is Brody over 25 years old?', 'Did Brody previously work as a developer?', 'Was Brody born in Washington?', 'Did Brody attend a CODE 102 course?', 'Is Brody enjoying CODE 201 so far?'];
 var strA = ['no', 'no', 'yes', 'yes', 'yes'];
 var input = '';
 var i = 0;
 var correct = 0;
+var g = 0;
+var q = 0;
 
-while(i < strQ.length) {
-  input = prompt(strQ[i]);
-  if (input.toLowerCase() === strA[i]) {
-    alert('Correct!');
-    correct++;
-  } else {
-    alert('Incorrect');
+function askYesNo(questionNo) {
+  while(i < strQ.length) {
+    input = prompt(strQ[i]);
+    if (input.toLowerCase() === strA[i]) {
+      alert('Correct!');
+      correct++;
+    } else {
+      alert('Incorrect');
+    }
+    q++;
+    i++;
   }
-  i++;
 }
 
-var numA = 24;
-
-var g = 0;
-var maxG = 4;
-
-input = Number(prompt('How old is Brody?' + ' You have ' + (maxG - g) + ' guesses left.'));
-if(input === numA) {
-  correct++;
+function askNumber(maxG) {
+  var numA = 24;
+  input = Number(prompt('How old is Brody?' + ' You have ' + (maxG - g) + ' guesses left.'));
   g++;
-  alert('Correct! It took you ' + g + ' guesses.');
-} else {
-  while(input !== numA) {
-    while(g < maxG) {
```

Comment:
Great idea getting rid of these nested while loops - not great performance. Glad you did conditionals within while loops instead.